### PR TITLE
Create snippet config extra request params

### PR DESCRIPTION
**Pull request by pmareke**
EX-4587

### CODE REVIEW

File: packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    public extraRequestParams: Dictionary<unknown> = {};
+
+    @Watch('snippetConfig', { deep: true, immediate: true })
+    syncExtraRequestParams({
+      lang,
+      searchLang,
+      instance,
+      scope,
+      ...extraRequestParams
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue:35]: What about the rest of the params like `currency`, `env`, `documentDirection`.... only the unknown parameters of the `snippetConfig` are considered extra. That's what I meant when I added the `     // Every other param` in the code I shared with you

File: packages/x-components/src/x-modules/extra-params/components/__tests__/snippet-config-extra-params.spec.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    const { wrapper } = renderSnippetConfigExtraParams();
+    const extraRequestParamsProvidedCallback = jest.fn();
+
+    wrapper.vm.$x
+      .on('ExtraRequestParamsProvided', true)
+      .subscribe(extraRequestParamsProvidedCallback);
+
+    expect(extraRequestParamsProvidedCallback).toHaveBeenCalledWith<
+      [WirePayload<Dictionary<unknown>>]
+    >({
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/extra-params/components/__tests__/snippet-config-extra-params.spec.ts:66]: I miss some new phase in this test modifying some snippetConfig value, and asserting that a new event is emitted. Also try modifying the SnippetConfig known parameters, and assert that no event is emitted.

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue (hunk 1 above)
Comment (pmareke) [reply packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue:35]: My fault.

File: packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  import Vue from 'vue';
+  import { Component, Watch } from 'vue-property-decorator';
+  import { xComponentMixin, XInject } from '../../../components';
+  import { Dictionary, forEach } from '../../../utils';
+  import { SnippetConfig } from '../../../x-installer';
+  import { extraParamsXModule } from '../x-module';
+  import ExtraParams from './extra-params.vue';
+
+  /**
+   * It renders a {@link ExtraParams} component and pass the snippetConfig.
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue:15]: You don't have to describe how this component does things, you have to describe what it does. Maybe something in the line of this?

> Extracts the extra parameters from the {@link SnippetConfig} and syncs it with the request objects of every x-module.

And I'm missing class properties/methods doc, and the `<docs>` block

**Review by davidmfempathy — APPROVED** [review]

File: packages/x-components/src/x-modules/extra-params/components/__tests__/snippet-config-extra-params.spec.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      metadata: { moduleName: 'extraParams' }
+    });
+
+    expect(extraParamsProvidedCallback).toHaveBeenCalledTimes(1);
+
+    await setSnippetConfig({ warehouse: 45678 });
+
+    expect(extraParamsProvidedCallback).toHaveBeenCalledWith<[WirePayload<Dictionary<unknown>>]>({
+      eventPayload: { warehouse: 45678 },
+      metadata: { moduleName: 'extraParams' }
```
Comment (javieri-empathy) [inline packages/x-components/src/x-modules/extra-params/components/__tests__/snippet-config-extra-params.spec.ts:80]: I'm still missing a test phase checking that if parameters like `lang`, `currency` that are not "extra", no event is emitted.

Comment (pmareke) [reply packages/x-components/src/x-modules/extra-params/components/__tests__/snippet-config-extra-params.spec.ts:80]: You're right, added it!

**Review by javieri-empathy — APPROVED** [review]

**Review by davidmfempathy — APPROVED** [review]

**Pull request merged**